src/emu/diimage.cpp

```diff
@@ -513,26 +496,32 @@ bool device_image_interface::load_software_region(const char *tag, optional_shar
 // to be loaded
 // ****************************************************************************
 
-void device_image_interface::run_hash(util::core_file &file, void (*partialhash)(util::hash_collection &, const unsigned char *, unsigned long, const char *),
-	util::hash_collection &hashes, const char *types)
+void device_image_interface::run_hash(util::core_file &file, uint32_t skip_bytes, util::hash_collection &hashes, const char *types)
 {
-	u32 size;
-	std::vector<u8> buf;
-
+	// reset the hash; we want to override existing data
 	hashes.reset();
-	size = (u32) file.size();
 
-	buf.resize(size);
-	memset(&buf[0], 0, size);
+	// figure out the size, and "cap" the skip bytes
+	uint64_t size = file.size();
+	skip_bytes = (uint32_t) std::min((uint64_t) skip_bytes, size);
 
-	// read the file
-	file.seek(0, SEEK_SET);
-	file.read(&buf[0], size);
+	// seek to the beginning
+	file.seek(skip_bytes, SEEK_SET);
+	uint64_t position = skip_bytes;
 
-	if (partialhash)
-		partialhash(hashes, &buf[0], size, types);
-	else
-		hashes.compute(&buf[0], size, types);
+	// keep on reading hashes
+	while(position < size)
```

Comment:
Eh?  This is what the code looks like... I'm not sure what the problem is?

	// seek to the beginning
	file.seek(skip_bytes, SEEK_SET);
	uint64_t position = skip_bytes;

	// keep on reading hashes
	while(position < size)
	{